### PR TITLE
Reuse docker image

### PR DESCRIPTION
**Pull request by troibe**
Depends on https://github.com/siliconcompiler/siliconcompiler/pull/1921
The @reuse-docker-image needs to be changed to @main before merging into main.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -16,33 +16,17 @@ env:
...
-        id: docker
-        run: |
-          pip install siliconcompiler
-          git -C siliconcompiler checkout v$(sc -version)
-          pip install -r siliconcompiler/setup/docker/requirements.txt
-          sc_tools_name=$(python3 siliconcompiler/setup/docker/builder.py --tool tools --registry ${{ env.REGISTRY }})
-          echo "sc_tools=${sc_tools_name}" >> $GITHUB_OUTPUT
+    uses: siliconcompiler/siliconcompiler/.github/workflows/docker_image.yml@reuse-docker-image
+    with:
+      tool: 'tools'
```
Comment (gadfort) [inline .github/workflows/ci.yml:21]: this needs the sc-version corrected.
```
pip install siliconcompiler
git -C siliconcompiler checkout v$(sc -version)
```

You probably need to add a pre-step to get this and use it

**Review by gadfort — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (troibe) [reply .github/workflows/ci.yml:21]: Hm yes that doesn't seem super pretty to me.
What do you think of having a `release: true` input or `version: latest` to move that logic into `docker_image.yml`?

Comment (troibe) [reply .github/workflows/ci.yml:21]: Note to myself should be possible without install using:
`git describe --abbrev=0 --tags`

Comment (gadfort) [reply .github/workflows/ci.yml:21]: `sc_version: latest` seems fine to me.

**Comment by troibe** [comment]
The `@latest-version-docker-image` needs to be changed to `@main` before merging into main.

**Comment by troibe** [comment]
Hope the version thing was actually the issue here...

**Review by gadfort — APPROVED** [review]

**Pull request merged**